### PR TITLE
feat(generative_ai): update gemini model to use flash model

### PR DESCRIPTION
**Pull request by msampathkumar**
## Description

Update Gemini Models to use flash

https://blog.google/technology/ai/google-gemini-update-flash-ai-assistant-io-2024/

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by m-strzelczyk — APPROVED** [review]

File: generative_ai/function_calling.py [hunk 1]
```diff
@@ -165,6 +165,7 @@ def generate_function_call_advanced(project_id: str) -> GenerationResponse:
         )
     )
 
+    # TODO(developer): Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/function_calling.py:168]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/gemini_all_modalities.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ def analyze_all_modalities(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # TODO(developer): Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_all_modalities.py:27]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/gemini_audio.py [hunk 3]
```diff
@@ -24,6 +24,7 @@ def summarize_audio(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # TODO(developer): Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_audio.py:27]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/gemini_audio.py [hunk 4]
```diff
@@ -55,6 +56,7 @@ def transcript_audio(project_id: str) -> str:
...
```
Comment (gericdong) [inline generative_ai/gemini_audio.py:59]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/gemini_count_token_example.py [hunk 5]
```diff
@@ -58,7 +58,8 @@ def count_tokens_multimodal(project_id: str) -> GenerationResponse:
 
     vertexai.init(project=project_id, location="us-central1")
 
-    model = GenerativeModel(model_name="gemini-1.5-pro-preview-0409")
+    # For Gemini supported models check https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_count_token_example.py:61]: Let's not put this comment. The page where the sample is shown should have the model information.

File: generative_ai/gemini_pdf_example.py [hunk 6]
```diff
@@ -24,6 +24,7 @@ def analyze_pdf(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # TODO(developer): Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_pdf_example.py:27]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/gemini_system_instruction.py [hunk 7]
```diff
@@ -24,6 +24,7 @@ def set_system_instruction(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # TODO(developer): Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_system_instruction.py:27]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/gemini_video_audio.py [hunk 8]
```diff
@@ -24,6 +24,7 @@ def analyze_video_with_audio(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # TODO(developer): Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_video_audio.py:27]: Let's not put the TODO here. The page where the sample is shown should have the model information.

File: generative_ai/function_calling.py [hunk 9]
```diff
@@ -165,6 +165,7 @@ def generate_function_call_advanced(project_id: str) -> GenerationResponse:
         )
     )
 
+    # Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/function_calling.py:168]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_all_modalities.py [hunk 10]
```diff
@@ -24,6 +24,7 @@ def analyze_all_modalities(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_all_modalities.py:27]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_audio.py [hunk 11]
```diff
@@ -24,6 +24,7 @@ def summarize_audio(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_audio.py:27]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_audio.py [hunk 12]
```diff
@@ -55,6 +56,7 @@ def transcript_audio(project_id: str) -> str:
...
```
Comment (gericdong) [inline generative_ai/gemini_audio.py:59]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_count_token_example.py (hunk 5 above)
Comment (gericdong) [inline generative_ai/gemini_count_token_example.py:61]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_pdf_example.py [hunk 13]
```diff
@@ -24,6 +24,7 @@ def analyze_pdf(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_pdf_example.py:27]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_system_instruction.py [hunk 14]
```diff
@@ -24,6 +24,7 @@ def set_system_instruction(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_system_instruction.py:27]: No need to add this link. The page where the sample is shown should have the model info.

File: generative_ai/gemini_video_audio.py [hunk 15]
```diff
@@ -24,6 +24,7 @@ def analyze_video_with_audio(project_id: str) -> str:
 
     vertexai.init(project=project_id, location="us-central1")
 
+    # Check latest AI models at https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference
```
Comment (gericdong) [inline generative_ai/gemini_video_audio.py:27]: No need to add this link. The page where the sample is shown should have the model info.

**Review by gericdong — APPROVED** [review]

**Pull request merged**